internal/kgateway/krtcollections/policy.go

```diff
@@ -228,7 +259,34 @@ func (i *BackendIndex) getBackendFromAlias(kctx krt.HandlerContext, gk schema.Gr
 	}
 
 	if out == nil {
-		return nil, &NotFoundError{NotFoundObj: key.ObjectSource}
+		// Check if any backend exists with the same service name but different port across all alias collections
+		serviceExists := false
+		objSource := key.ObjectSource
+		for _, actualGk := range actualGks {
+			col, ok := i.availableBackends[actualGk]
+			if !ok {
+				continue
+			}
+
+			backends := krt.Fetch(kctx, col, krt.FilterGeneric(func(obj interface{}) bool {
```

Comment:
```suggestion
			backends := krt.Fetch(kctx, col, krt.FilterGeneric(func(obj ir.BackendObjectIR) bool {
```

We should also do the fetch once like we do [here](https://github.com/kgateway-dev/kgateway/blob/main/internal/kgateway/krtcollections/policy.go#L174). Let's have `getBackendFromAlias` do the check  if any backend exists with the same service name but different port across all alias collections instead of repeating the same logic twice.